concrete-core/src/backends/core/private/math/random/generator.rs

```diff
@@ -65,8 +71,15 @@ impl RandomGenerator {
     /// let mut generator = RandomGenerator::new(None);
     /// assert!(!generator.is_bounded());
     /// ```
-    pub fn new(seed: Option<u128>) -> RandomGenerator {
-        RandomGenerator(RandomGeneratorImpl::new(seed))
+    pub fn new(seed: Option<u128>) -> RandomGenerator<G> {
+        let seed = seed.map(Seed).unwrap_or_else(|| {
+            if !is_x86_feature_detected!("rdseed") {
+                RdseedSeeder.seed()
+            } else {
+                LinuxSeeder::new(0u128).seed()
```

Comment:
Here if the user is not using an x86 processor and is not on Linux, this won't work, right? We should probably handle it in a better way?

Reply:
Also, line 76 why is there a `!is_x86_feature_detected`, shouldn't it be the opposite?